configure.py

```diff
@@ -904,9 +898,11 @@ def pkg_config(pkg):
     try:
       proc = subprocess.Popen(shlex.split(pkg_config) + args,
                               stdout=subprocess.PIPE)
-      val = to_utf8(proc.communicate()[0]).strip()
+      with proc:
```

Comment:
I have never seen this but it is cool if it works.  Perhaps we should upgrade to [`subprocess.run()`](https://docs.python.org/3/library/subprocess.html) which simplifies this dance.

Reply:
It does work, though I don't know how we can change this to `subprocess.run()`, may you show how?